### DISCUSSION TITLE
Advantage of using sub-stream?

### DISCUSSION THREAD

**Discussion by cwleth** [blakeblackshear/frigate#1090] [Ask A Question]
What is the advantage of using sub- & main stream (with detection on sub & clip on main) vs. everything on mainstream only for person detection?

Sub: 640x360
Main: 1920x1080

![image](https://user-images.githubusercontent.com/78813007/118047731-7af29680-b37b-11eb-940b-1138189daef4.png)

**Comment by paularmstrong** [comment]
Using the smaller substream will reduce:
* CPU load and/or processing speed
* Network usage

**Reply by blackgold9** [reply]
Is there an understanding of how substream benefits trade off with detection accuracy? In my (albeit limited) testing, switching from detecting on my 480p substream to the full 4k helped with detecting smaller targets. I do still wonder though if there is some other level i could pull to improve detection rate using the lighter weight stream.

**Comment by trythatagain** [comment]
When I first started playing around with Frigate, I found the size of a 300*300 block on the sub stream (704 * 576) was just too big, so I use the main stream (3072 * 1728) for detection. I believe it means I can detect smaller objects.

**Comment by haggercody** [comment]
(on beta here where i think the behavior is a bit different with recording) i notice there is an ffmpeg job to match all the streams i am using;  Granted the main streams are more cpu, but it seems like if i don't use the main stream i have less overall ffmpeg jobs running?  My guess is the detect job costs more on a main stream vs sub?

**Comment by rhatguy** [comment]
Correct.  The record streams are just written directly to disk and not decoded and processed.  The detect streams have to be first decoded (hopefully in hardware), then have motion detection run (on the CPU), then potentially have object detection run (hopefully on a coral or other hardware).  If you run frigates "detect" pipeline on your cameras main stream, all of those operations are going to have to be done on a larger video file which will expend more resources.

The general idea is to limit the full pipeline (decoding, motion detection, object detection) to only running on the substream.  Then if something triggers there you "record" the main stream so you have the best quality video of events.

**Reply by haggercody** [reply]
also i had one of my streams resolution set incorrectly in the config and it looked like it was causing more cpu usage--my guess is maybe it was getting resized.

**Reply by blakeblackshear** [reply]
> also i had one of my streams resolution set incorrectly in the config and it looked like it was causing more cpu usage--my guess is maybe it was getting resized.

0.9.0 does this rather than error out now, so that is expected
